docs/source/example_notebooks/dowhy_refuter_assess_overlap.ipynb

```diff
@@ -0,0 +1,508 @@
+    "jitter_data['X2'] = jitter_data['X2'] + rng.normal(0, 0.05, size=(200,)) \n",
+    "\n",
+    "fig, ax = plt.subplots(1, 1, figsize=(5, 5))\n",
+    "for t in [0, 1]:\n",
+    "    this_data = jitter_data.query('T == @t')\n",
+    "    ax.scatter(this_data['X1'], this_data['X2'], label=f'T = {t}', edgecolor='w', linewidths=0.5)\n",
+    "\n",
+    "plt.legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11abab7e",
+   "metadata": {},
+   "source": [
+    "Notice how the only region where overlap holds is when $X_1 = X_2 = 0$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6a5d5c53",
+   "metadata": {},
+   "source": [
+    "## Using OverRule as a refuter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa1618d9",
+   "metadata": {},
+   "source": [
+    "This method is currently implemented as a refutation method, though it can also be accessed through the functional API.  We illustrate both methods below, before discussing their arguments.\n",
+    "\n",
+    "Note that OverRule returns rules in Disjunctive Normal Form (DNF), which can be read as an OR of ANDs, and are given when `print(refute)` is called."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e165ecb",
+   "metadata": {},
+   "source": [
+    "### Example usage using CausalModel"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cb218c53",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = CausalModel(\n",
+    "    data=test_data,\n",
+    "    treatment=\"T\",\n",
+    "    outcome=\"Y\",\n",
+    "    common_causes=['X1', 'X2']\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "d1b9d144",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "identified_estimand = model.identify_effect(proceed_when_unidentifiable=True)\n",
+    "\n",
+    "estimate = model.estimate_effect(identified_estimand, method_name=\"backdoor.linear_regression\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "8b0e895d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We disucss configuration in more depth below.\n",
+    "support_config = SupportConfig(seed=0)\n",
+    "overlap_config = OverlapConfig()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "e6afb904",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "refute = model.refute_estimate(\n",
+    "    identified_estimand, \n",
+    "    estimate, \n",
+    "    method_name='assess_overlap', \n",
+    "    support_config=support_config, \n",
+    "    overlap_config=overlap_config,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "e11e8d3b",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Rules cover 50.0% of all samples\n",
```

Comment:
The interpretation for support and overlap rules is the same here, since they are both DNF.  A data point is eligible if *any* of the support rules and *any* of the overlap rules apply.

Note that, in part to make it easier to review the code & standardize the output, I've restricted to learning only DNF rules in this PR - the example from the paper you're referring to uses CNF rules for the output.  

I added clarifications on this in the output itself, and in the notebook.